apps/frontend/src/pages/home/components/feedbacks-section/components/feedback-modals/add-feedback.modal.tsx

```diff
@@ -0,0 +1,101 @@
+import { useCallback, useState } from "react";
+import { toast } from "react-toastify";
+
+import { Button, Input, Loader } from "~/libs/components/components.js";
+import { getClassNames } from "~/libs/helpers/get-class-names.js";
+import { useAppForm } from "~/libs/hooks/hooks.js";
+import {
+	feedbackApi,
+	type FeedbackCreateRequestDto,
+	feedbackCreateValidationSchema,
+	FeedbackValidationRule,
+} from "~/modules/feedbacks/feedbacks.js";
+
+import styles from "./styles.module.css";
+
+type Properties = {
+	onClose: () => void;
+	userId?: number;
+};
+
+const AddFeedbackModal: React.FC<Properties> = ({
+	onClose,
+	userId,
+}: Properties) => {
+	const [isLoading, setIsLoading] = useState<boolean>(false);
+	const { control, errors, handleSubmit, watch } =
+		useAppForm<FeedbackCreateRequestDto>({
+			defaultValues: { text: "", userId: Number(userId) },
+			validationSchema: feedbackCreateValidationSchema,
+		});
+
+	const textValue = watch?.("text") ?? "";
+	const characterCount = textValue.length;
+	const maxCharacters = FeedbackValidationRule.TEXT_MAX_LENGTH;
+
+	const handleFormSubmit = useCallback(
+		async (payload: FeedbackCreateRequestDto): Promise<void> => {
+			setIsLoading(true);
+
+			try {
+				await feedbackApi.create(payload);
+				toast.success("Feedback was successfully added!");
+				onClose();
+			} catch {
+				toast.error("Failed to add feedback. Please try again later.");
+			} finally {
+				setIsLoading(false);
+			}
+		},
+		[onClose],
+	);
+
+	const handleOnSubmit = useCallback(
+		(event: React.FormEvent<HTMLFormElement>) => {
+			void handleSubmit(handleFormSubmit)(event);
+		},
+		[handleSubmit, handleFormSubmit],
+	);
+
+	return (
+		<form
+			aria-labelledby="feedback-title"
+			className={getClassNames(styles["form"], "cluster")}
+			noValidate
+			onSubmit={handleOnSubmit}
+		>
+			<h2 className={styles["feedback-title"]} id="feedback-title">
+				Add your feedback
+			</h2>
+			<Input
+				control={control}
+				errors={errors}
+				hasLabel={false}
+				isRequired
+				isTextArea
+				label="Testimonial"
+				name="text"
+				placeholder="Enter your testimonial"
+				type="text"
+			/>
```

Comment:
```suggestion
			<Input
				control={control}
				errors={errors}
				hasLabel={false}
				isRequired
				isTextArea
				label="Testimonial"
				name="text"
				placeholder="Enter your testimonial"
				type="text"
			/>
```
This is wrong, the Input component can not have `isTextArea` property as it is a totally separate entity. Create a separate resuable component for that and use it here